sdk/security_keyvault/src/clients/keyvault_client.rs

```diff
@@ -54,63 +58,51 @@ impl KeyvaultClient {
     ) -> azure_core::Result<Self> {
         let vault_url = Url::parse(vault_url)?;
         let endpoint = extract_endpoint(&vault_url)?;
+        let pipeline = new_pipeline_from_options(token_credential.clone(), endpoint.clone());
         let client = Self {
             vault_url,
             endpoint,
-            token_credential,
-            token: None,
+            pipeline,
         };
         Ok(client)
     }
 
-    pub(crate) async fn request(
-        &mut self,
-        method: reqwest::Method,
-        uri: String,
-        body: Option<String>,
-    ) -> azure_core::Result<String> {
-        self.get_token().await?;
+    pub(crate) fn finalize_request(
+        &self,
+        mut url: Url,
+        method: Method,
+        headers: Headers,
+        request_body: Option<Body>,
+    ) -> azure_core::Result<Request> {
+        let dt = OffsetDateTime::now_utc();
+        let time = date::to_rfc1123(&dt);
+
+        url.set_query(Some(API_VERSION_PARAM));
```

Comment:
This makes https://github.com/Azure/azure-sdk-for-rust/pull/1011 no longer necessary.